src/scripts/main.js

```diff
@@ -1,3 +1,371 @@
+/* eslint-disable no-shadow */
+/* eslint-disable max-len */
 'use strict';
 
-// write your code here
+const startBtn = document.querySelector('.start');
+const gameBoard = document.querySelectorAll('.field-cell');
+const score = document.querySelector('.game-score');
+const messageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+
+let scoreCounter = 0;
+let hasButtonBeenClicked = false;
+let stopGame = false;
+
+const field = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+
+function randomTile() {
+  const randomNumber = Math.floor(Math.random() * 16);
+
+  const fieldWithoutZero = field.every(n => n !== 0);
+
+  if (fieldWithoutZero) {
+    return;
+  }
+
+  if (field[randomNumber] !== 0) {
+    return randomTile();
+  }
+
+  field[randomNumber] = Math.random() < 0.1 ? 4 : 2;
+
+  gameBoard[randomNumber].classList.add(`field-cell--${field[randomNumber]}`);
+  gameBoard[randomNumber].innerHTML = field[randomNumber];
+
+  checkAdjacentFields();
+}
+
+function start() {
+  startBtn.addEventListener('click', () => {
+    if (hasButtonBeenClicked) {
+      location.reload();
```

Comment:
would be better to reset your game field instead of page reload (just my opinion)